rs/execution_environment/src/canister_manager.rs

```diff
@@ -2059,7 +2059,8 @@ impl CanisterManager {
                     Some(canister_state) => {
                         if !canister_state.controllers().contains(&sender) {
                             return (
-                                Err(CanisterManagerError::CanisterSnapshotInvalidOwnership {
+                                Err(CanisterManagerError::CanisterSnapshotNotController {
+                                    sender,
                                     canister_id,
```

Comment:
Wouldn't it make more sense to pass the `snapshot_canister_id` here and update the textual error message? The `canister_id` is directly present in the input payload (so that information is merely reflected here), but the `snapshot_canister_id` might not be apparent to the user.